lib/foreman/cast.rb

```diff
@@ -0,0 +1,26 @@
+module Foreman
+  module Cast
+    def self.to_bool(value)
+      case value
+
+      when String
+        return true if value =~ (/\A(true|t|yes|y|on|1)\z/i)
+        return false if value.blank? || value =~ (/\A(false|f|no|n|off|0)\z/i)
```

Comment:
It wasn't a comment based on the language, but rather how this method is used.  Often we differentiate between a nil/empty and a true/false value.

As an example, if I perform an update against a setting with no value (which is effectively nil), then the value is being changed to false.

    GET /api/v2/settings/Parametrized_Classes_in_ENC
    value is true

    PUT /api/v2/settings/Parametrized_Classes_in_ENC
    {"setting":{}}

    GET /api/v2/settings/Parametrized_Classes_in_ENC
    value is false